### DISCUSSION TITLE
Add icon with command to Terminal toolbar?

### DISCUSSION THREAD

**Discussion by msweeney1999** [microsoft/vscode-discussions#991] [Extension Development QnA] [answered]
My extension creates a custom terminal using the vscode.window.createTerminal API.  Is it possible to add an icon/command to this terminal?
Thanks

![toolbar](https://github.com/microsoft/vscode-discussions/assets/19656050/f5a9fa47-721f-41ac-a2f8-76294dc69241)

**Comment by alefragnani** [comment]
Hi @msweeney1999 ,

For sure you can. If you want to change the `powershell` icon on your terminal, you could use the `iconPath` attribute in `TerminalOptions` or `ExtensionTerminalOptions` and it works just fine. Something like this:

```ts
    const terminal = vscode.window.createTerminal({ name: `My Extension REPL`, pty, iconPath: new vscode.ThemeIcon("zap") });
```

![image](https://github.com/microsoft/vscode-discussions/assets/3781424/16fbd804-91ab-4e79-b95f-d0e11ad064b9)

On the other hand, if you want add icons to the _terminal name_, in the location of the image, you could use `codicons` in the terminal name. Something like this:

```ts
    const terminal = vscode.window.createTerminal({ name: `My Extension REPL $(zap)`, pty});
```

![image](https://github.com/microsoft/vscode-discussions/assets/3781424/80eaf67a-f0d5-4f30-92b0-7a7d561a0e25)

For a full reference of codicons, look at https://microsoft.github.io/vscode-codicons/dist/codicon.html

Hope this helps

ps: sorry for the updated answer. For some reason Github wasn't showing your image when I loaded the message and posted the original answer

**Reply by msweeney1999** [reply]
Thanks Alessandro,

But how do I attach a command to the icon?

Mark

**Reply by alefragnani** [reply, marked as answer]
Hi @msweeney1999 ,

You can’t assign commands to that element, like a `TreeItem`. It is just a title that you define to the terminal. 

The two commands you see on each terminal title (split, delete) are provided by VS Code, and there is no API, yet, to add new commands.

hope this helps

**Reply by msweeney1999** [reply]
Understood.  Thanks for the feedback.
Mark
